dotnet/Services/ProductReviewService.cs

```diff
@@ -334,25 +334,19 @@ public async Task<ReviewsResponseWrapper> GetReviewsByProductId(string productId
                 searchQuery = $"{searchQuery}&approved=true";
             }
 
-            if (pastReviews)
+            if (pastReviews && !string.IsNullOrEmpty(locale))
             {
-                if (rating == 0) {
-                    wrapper = await this._productReviewRepository.GetProductReviewsMD($"productId={productId}{sort}{searchQuery}", from.ToString(), to.ToString());
+                string productQuery = $" AND productId={productId}";
+                if (rating > 0 && rating <= 5)
+                {
+                    ratingQuery = $" AND rating={rating}";
                 }
-                else
+                if (!string.IsNullOrEmpty(locale))
                 {
-                    string productQuery = $" AND productId={productId}";
-                    if (rating > 0 && rating <= 5)
-                    {
-                        ratingQuery = $" AND rating={rating}";
-                    }
-                    if (!string.IsNullOrEmpty(locale))
-                    {
-                        localeQuery = $"((locale=*{locale}-*) OR (locale is null))";
-                    }
-
-                    wrapper = await this._productReviewRepository.GetProductReviewsMD($"_where={localeQuery}{ratingQuery}{productQuery}{searchQuery}{sort}", from.ToString(), to.ToString());
+                    localeQuery = $"((locale=*{locale}-*) OR (locale is null))";
```

Comment:
To improve readability, I would move this up to line 339, so that `localeQuery` is given a value before `productQuery` and `ratingQuery`. In other words, set their values in the same order that they are added here: `$"_where={localeQuery}{ratingQuery}{productQuery}{searchQuery}{sort}"`